### PR TITLE
Add support for 8 character hex codes

### PR DESCRIPTION
**Pull request by benmccann**
Copies the latest code from upstream to fix https://github.com/chartjs/Chart.js/issues/6035

### CODE REVIEW

File: color-string.js [hunk 1]
```diff
@@ -24,59 +24,81 @@ function getRgba(string) {
...
-       rgba = /^rgba?\(\s*([+-]?\d+)\s*,\s*([+-]?\d+)\s*,\s*([+-]?\d+)\s*(?:,\s*([+-]?[\d\.]+)\s*)?\)$/i,
-       per = /^rgba?\(\s*([+-]?[\d\.]+)\%\s*,\s*([+-]?[\d\.]+)\%\s*,\s*([+-]?[\d\.]+)\%\s*(?:,\s*([+-]?[\d\.]+)\s*)?\)$/i,
-       keyword = /(\w+)/;
-
-   var rgb = [0, 0, 0],
-       a = 1,
-       match = string.match(abbr);
-   if (match) {
+
+   var abbr = /^#([a-f0-9]{3,4})$/i;
```
Comment (kurkle) [inline color-string.js]: /i not needed here

File: color-string.js [hunk 2]
```diff
@@ -24,59 +24,81 @@ function getRgba(string) {
...
+   var hex = /^#([a-f0-9]{6})([a-f0-9]{2})?$/i;
+   var rgba = /^rgba?\(\s*([+-]?\d+)\s*,\s*([+-]?\d+)\s*,\s*([+-]?\d+)\s*(?:,\s*([+-]?[\d\.]+)\s*)?\)$/;
```
Comment (kurkle) [inline color-string.js:30]: `/i` here makes a difference

File: color-string.js [hunk 3]
```diff
@@ -24,59 +24,81 @@ function getRgba(string) {
...
+   var per = /^rgba?\(\s*([+-]?[\d\.]+)\%\s*,\s*([+-]?[\d\.]+)\%\s*,\s*([+-]?[\d\.]+)\%\s*(?:,\s*([+-]?[\d\.]+)\s*)?\)$/;
+   var keyword = /(\D+)/;
```
Comment (kurkle) [inline color-string.js:32]: Why do we need to change `keyword` (and why was the change needed to `w+` in the first place)?

File: test/basic.js [hunk 4]
```diff
@@ -33,6 +36,9 @@ assert.equal(string.getAlpha("hwb(244, 100%, 100%, 0.6)"), 0.6);
 assert.equal(string.getAlpha("hwb(244, 100%, 100%)"), 1);
 
 // alpha
+assert.deepEqual(string.getRgba('#c814e933'), [200, 20, 233, 0.2]);
+assert.deepEqual(string.getRgba('#c814e900'), [200, 20, 233, 0]);
+assert.deepEqual(string.getRgba('#c814e9ff'), [200, 20, 233, 1]);
 assert.deepEqual(string.getRgba("rgba(200, 20, 233, 0.2)"), [200, 20, 233, 0.2]);
 assert.deepEqual(string.getRgba("rgba(200, 20, 233, 0)"), [200, 20, 233, 0]);
 assert.deepEqual(string.getRgba("rgba(100%, 30%, 90%, 0.2)"), [255, 77, 229, 0.2]);
```
Comment (kurkle) [inline test/basic.js:44]: Should add a test for "RGBA(...)" as that should be supported.

**Review by kurkle — CHANGES_REQUESTED** [review]
Should also update `README.md`
Might be easier to rebase?

**Comment by simonbrunel** [comment]
Duplicated of #3, which I'm not sure why it hasn't been merged yet. I prefer #3 since it doesn't change the surrounding code, thus easier to review.

**Pull request closed without merging**